app/controllers/users_controller.rb

```diff
@@ -17,15 +17,11 @@ def new
 
   def create
     @user = User.new(user_params)
-    respond_to do |format|
-      if @user.save
-        session[:user_id] = @user.id
-        format.html { redirect_to root_path, notice: 'User was successfully created.' }
-        format.json { render :index, status: :created, location: @user }
-      else
-        format.html { render :new }
-        format.json { render json: @user.errors, status: :unprocessable_entity }
-      end
```

Comment:
Not entirely sure why the json was taken out here as part of this branch, but it looks OK so I don't see a reason to not take it out.